### PR TITLE
Add Lock Support to SwitchBot Cloud Integration

### PR DESCRIPTION
**Pull request by miterion**
## Proposed change

This PR adds docs for the added SwitchBot Lock support.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/115128.
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Enhanced integration documentation for SwitchBot devices to include support for Lock devices.
	- Added "Lock" to the categories and platforms sections, expanding the range of supported devices.
	- Updated the supported devices section to explicitly mention Lock devices.

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
Oh hi there @miterion 👋 

There is a merge conflict on this PR. Could you take a look and resolve it? Also, don't forget to mark this PR ready for review once you've fixed the conflict.

Thanks 👍 

../Frenck

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @miterion :+1:

File: source/_integrations/switchbot_cloud.markdown [hunk 1]
```diff
@@ -44,6 +46,7 @@ Please note, device names configured in the SwitchBot app are transferred into H
 - Meter
 - MeterPlus
 - Outdoor Meter
+- Lock
```
Comment (c0ffeeca7) [inline source/_integrations/switchbot_cloud.markdown]:
```suggestion
```

File: source/_integrations/switchbot_cloud.markdown [hunk 2]
```diff
@@ -44,6 +46,7 @@ Please note, device names configured in the SwitchBot app are transferred into H
 - Meter
```
Comment (c0ffeeca7) [inline source/_integrations/switchbot_cloud.markdown:47]:
```suggestion
- Lock
- Meter
```

File: source/_integrations/switchbot_cloud.markdown [hunk 3]
```diff
@@ -18,6 +19,7 @@ ha_domain: switchbot_cloud
 ha_platforms:
   - climate
   - sensor
+  - lock
```
Comment (c0ffeeca7) [inline source/_integrations/switchbot_cloud.markdown]:
```suggestion
  - lock
  - sensor
```

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**